keras_nlp/src/models/retinanet/non_max_supression.py

```diff
@@ -0,0 +1,623 @@
+# limitations under the License.
+
+import math
+
+import keras
+from keras import ops
+
+from keras_nlp.src.bounding_box import converters
+from keras_nlp.src.bounding_box import utils
+from keras_nlp.src.bounding_box import validate_format
+
+try:
+    import tensorflow as tf
+    import torchvision
+except ImportError:
+    tf = None
+    torchvision = None
+
+EPSILON = 1e-8
+
+
+class NonMaxSuppression(keras.layers.Layer):
+    """A Keras layer that decodes predictions of an object detection model.
+
+    Args:
+      bounding_box_format: The format of bounding boxes of input dataset. Refer
+        [to the keras.io docs](https://keras.io/api/keras_cv/bounding_box/formats/)
+        for more details on supported bounding box
+        formats.
+      from_logits: boolean, True means input score is logits, False means
+        confidence.
+      iou_threshold: a float value in the range [0, 1] representing the minimum
+        IoU threshold for two boxes to be considered same for suppression.
+        Defaults to 0.5.
+      confidence_threshold: a float value in the range [0, 1]. All boxes with
+        confidence below this value will be discarded, defaults to 0.5.
+      max_detections: the maximum detections to consider after nms is applied. A
+        large number may trigger significant memory overhead, defaults to 100.
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        bounding_box_format,
+        from_logits,
+        iou_threshold=0.5,
+        confidence_threshold=0.5,
+        max_detections=100,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.bounding_box_format = bounding_box_format
+        self.from_logits = from_logits
+        self.iou_threshold = iou_threshold
+        self.confidence_threshold = confidence_threshold
+        self.max_detections = max_detections
+        self.built = True
+
+    def call(
+        self, box_prediction, class_prediction, images=None, image_shape=None
+    ):
+        """Accepts images and raw predictions, and returns bounding box
+        predictions.
+
+        Args:
+            box_prediction: Dense Tensor of shape [batch, boxes, 4] in the
+                `bounding_box_format` specified in the constructor.
+            class_prediction: Dense Tensor of shape [batch, boxes, num_classes].
+        """
+        target_format = "yxyx"
+        if utils.is_relative(self.bounding_box_format):
+            target_format = utils.as_relative(target_format)
+
+        box_prediction = converters.convert_format(
+            box_prediction,
+            source=self.bounding_box_format,
+            target=target_format,
+            images=images,
+            image_shape=image_shape,
+        )
+        if self.from_logits:
+            class_prediction = ops.sigmoid(class_prediction)
+
+        confidence_prediction = ops.max(class_prediction, axis=-1)
+
+        if keras.backend.backend() == "tensorflow":
```

Comment:
I think we should remove all of this backend switch casing. Let's focus on Jax as the defacto backbone, and right things with `keras.ops` everywhere.

OK to do something like [this](https://github.com/keras-team/keras-nlp/blob/0472bcbb439a9813c8dbec2e99d73d31d1b5b3b0/keras_nlp/src/layers/modeling/masked_lm_head.py#L178-L187) to avoid a bug on the tf backend, but let's avoid totally forking the implementation.

We can do that as a follow up for efficiency if we want. But we would want to do that by adding some more ops for this into core Keras, and only if the naive ops implementation + XLA is significantly slower. Definitely better done as a follow up.

Reply:
Removed backend switching, all backends should work without switch logic as well.